src/java/org/apache/cassandra/net/AbstractMessageHandler.java

```diff
@@ -304,12 +302,21 @@ private void onReserveCapacityRegained(Limit endpointReserve, Limit globalReserv
         try
         {
             /*
-             * Process up to one message using supplied overriden reserves - one of them pre-allocated,
-             * and guaranteed to be enough for one message - then, if no obstacles enountered, reactivate
+             * Process up to one message using supplied overridden reserves - one of them pre-allocated,
+             * and guaranteed to be enough for one message - then, if no obstacles encountered, reactivate
              * the frame decoder using normal reserve capacities.
              */
-            if (processUpToOneMessage(endpointReserve, globalReserve))
-                decoder.reactivate();
+            if (!decoder.isActive())
```

Comment:
Why has this condition been added? This method should only be invoked on connections that have been paused due to breaching their memory limits?

Reply:
It does look unnecessary. Removing.